src/Compilers/Core/Portable/CodeGen/ILBuilder.cs

```diff
@@ -70,7 +70,7 @@ internal ILBuilder(ITokenDeferral module, LocalSlotManager localSlotManager, Opt
 
             leaderBlock = _currentBlock = _scopeManager.CreateBlock(this);
 
-            _labelInfos = new SmallDictionary<object, LabelInfo>(ReferenceEqualityComparer.Instance);
+            _labelInfos = new SmallDictionary<object, LabelInfo>();
```

Comment:
Also, I have no idea what does "Label declarations and references are distinct instances" mean. This doesn't make any sense to me. Let's talk offline.